src/sentry/rules/processing/delayed_processing.py

```diff
@@ -463,6 +473,55 @@ def bucket_num_groups(num_groups: int) -> str:
     return "1"
 
 
+def process_rulegroups_in_batches(project_id: int, batch_size=CHUNK_BATCH_SIZE):
+    """
+    This will check the number of rulegroup_to_event_data items in the Redis buffer for a project.
+
+    If the number is larger than the batch size, it will chunk the items and process them in batches.
+
+    The batches are replicated into a new redis hash with a unique filter (a uuid) to identify the batch.
+    We need to use a UUID because these batches can be created in multiple processes and we need to ensure
+    uniqueness across all of them for the centralized redis buffer. The batches are stored in redis because
+    we shouldn't pass complex objects in the celery task arguments, and we can't send a page of data in the
+    batch becaues redis may not maintain the sort of the hash response.
+
+    `apply_delayed` will fetch the batch from redis and process the rules.
+    """
+    event_count = buffer.backend.get_hash_length(Project, {"project_id": project_id})
+
+    if event_count < batch_size:
+        return apply_delayed.delayed(project_id)
+
+    logger.info(
+        "delayed_processing.process_large_batch",
+        extra={"project_id": project_id, "count": event_count},
+    )
+
+    # if the dictionary is large, get the items and chunk them.
+    rulegroup_to_event_data = fetch_rulegroup_to_event_data(project_id)
+
+    with metrics.timer("delayed_processing.process_batch.duration"):
+        items = iter(rulegroup_to_event_data.items())
+
+        while batch := dict(islice(items, batch_size)):
+            batch_key = str(uuid.uuid4())
+
+            for field, value in batch.items():
+                buffer.backend.push_to_hash(
```

Comment:
looking at a way to do this in bulk rn - i am prreeettty sure that `redis.hset` supports this (since we're just setting a key / value on the hash). 🔬